src/components/tooltip/tooltip.test.js

```diff
@@ -0,0 +1,98 @@
+import Tooltip, { insertZeroWidthSpace } from './index';
+import { setup } from '../../utils/state.mock';
+
+const mockProps = {
+  chartSize: {
+    height: 766,
+    left: 0,
+    outerHeight: 766,
+    outerWidth: 1198,
+    sidebarWidth: 300,
+    top: 0,
+    width: 898
+  },
+  targetRect: {
+    bottom: 341.05895,
+    height: 2.1011,
+    left: 856.19622,
+    right: 866.03076,
+    top: 338.95785,
+    width: 9.83453,
+    x: 856.19622,
+    y: 338.95785
+  },
+  text: 'lorem_ipsum-dolor: sit [amet]',
+  visible: true
+};
+
+describe('Tooltip', () => {
+  it('renders without crashing', () => {
+    const wrapper = setup.shallow(Tooltip);
+    const container = wrapper.find('.pipeline-tooltip');
+    expect(container.length).toBe(1);
+  });
+
+  it('should not add the top class when the tooltip is towards the bottom', () => {
+    const targetRect = {
+      ...mockProps.targetRect,
+      top: mockProps.chartSize.height - 10
+    };
+    const wrapper = setup.shallow(Tooltip, { ...mockProps, targetRect });
+    const container = wrapper.find('.pipeline-tooltip--top');
+    expect(container.length).toBe(0);
+  });
+
+  it('should not add the right class when the tooltip is towards the left', () => {
+    const targetRect = {
+      ...mockProps.targetRect,
+      left: 10
+    };
+    const wrapper = setup.shallow(Tooltip, { ...mockProps, targetRect });
+    const container = wrapper.find('.pipeline-tooltip--right');
+    expect(container.length).toBe(0);
+  });
+
+  it("should add the 'top' class when the tooltip is towards the top", () => {
+    const targetRect = {
+      ...mockProps.targetRect,
+      top: 10
+    };
+    const wrapper = setup.shallow(Tooltip, { ...mockProps, targetRect });
+    const container = wrapper.find('.pipeline-tooltip--top');
+    expect(container.length).toBe(1);
+  });
+
+  it("should add the 'right' class when the tooltip is towards the right", () => {
+    const targetRect = {
+      ...mockProps.targetRect,
+      left: mockProps.chartSize.width - 10
+    };
+    const wrapper = setup.shallow(Tooltip, { ...mockProps, targetRect });
+    const container = wrapper.find('.pipeline-tooltip--right');
+    expect(container.length).toBe(1);
+  });
+});
+
+describe('insertZeroWidthSpace', () => {
+  it('wraps special characters with zero-width spaces', () => {
+    expect(insertZeroWidthSpace('-').length).toBe(3);
```

Comment:
I think it's better to compare with the exact expected result string for these rather than just length, unless there was a reason?

Reply:
Basically, the expected string result contains zeroWidthSpaces, which don't show up in  the text editor in a string. I can however use the same trick for this that I did in the function itself  - that is, this: 
```JavaScript
const zeroWidthSpace = String.fromCharCode(0x200b);

expect(text).toBe(`${zeroWidthSpace}${text}${zeroWidthSpace}`);
```

I'll change it now. Thanks for the review!

Reply:
@bru5 Fixed. Thanks!